node/src/accountant/scanners/scan_schedulers.rs

```diff
@@ -538,15 +553,76 @@ mod tests {
         let now = UNIX_EPOCH
             .checked_add(Duration::from_secs(1_000_000))
             .unwrap();
-        let subject = NewPayableScanDynIntervalComputerReal::default();
+        let mut subject = make_subject();
+        subject.clock = Box::new(SimpleClockMock::default().now_result(now));
+        subject.last_scan_timestamp = now.checked_add(Duration::from_secs(1)).unwrap();
 
-        let _ = subject.compute_interval(
-            now.checked_sub(Duration::from_secs(1)).unwrap(),
-            now,
-            Duration::from_secs(32),
+        let _ = subject.compute_interval();
+    }
+
+    #[test]
+    fn zero_out_works_for_default_subject() {
+        let mut subject = make_subject();
+        let last_scan_timestamp_before = subject.last_scan_timestamp;
+        let before_act = SystemTime::now();
+
+        subject.zero_out();
+
+        let after_act = SystemTime::now();
+        let last_scan_timestamp_after = subject.last_scan_timestamp;
+        assert_eq!(last_scan_timestamp_before, UNIX_EPOCH);
+        assert!(
+            before_act <= last_scan_timestamp_after && last_scan_timestamp_after <= after_act,
+            "we expected the last_scan_timestamp to be reset to now, but it was not"
         );
     }
 
+    #[test]
+    fn zero_out_works_for_general_subject() {
+        let mut subject = make_subject();
+        subject.last_scan_timestamp = SystemTime::now()
+            .checked_sub(Duration::from_secs(100))
+            .unwrap();
+        let before_act = SystemTime::now();
+
+        subject.zero_out();
+
+        let after_act = SystemTime::now();
+        let last_scan_timestamp_after = subject.last_scan_timestamp;
+        assert!(
+            before_act <= last_scan_timestamp_after && last_scan_timestamp_after <= after_act,
+            "we expected the last_scan_timestamp to be reset to now, but it was not"
+        );
+    }
+
+    #[test]
+    fn update_last_new_payable_scan_timestamp_works() {
+        let zero_out_params_arc = Arc::new(Mutex::new(vec![]));
+        let scan_intervals = ScanIntervals::compute_default(TEST_DEFAULT_CHAIN);
+        let mut subject = ScanSchedulers::new(scan_intervals, true);
+        subject.payable.dyn_interval_computer = Box::new(
+            NewPayableScanDynIntervalComputerMock::default().zero_out_params(&zero_out_params_arc),
+        );
+
+        subject.payable.update_last_new_payable_scan_timestamp();
+
+        let zero_out_params = zero_out_params_arc.lock().unwrap();
+        assert_eq!(*zero_out_params, vec![()])
+    }
+
+    fn make_subject() -> NewPayableScanDynIntervalComputerReal {
```

Comment:
Migrate these functions to the test_utils file.

Reply:
Only code that is general enough that it finds usages in more than a single file belongs there. 

I'm not going to flip this code into something public while it can be safely private and just local.